### PR TITLE
Implements FlatButton debugFillProperties

### PR DESCRIPTION
**Pull request by konifar**
ref: https://github.com/flutter/flutter/issues/11808

## Overview
Implemented FlatButton debugFillProperties.

### CODE REVIEW

File: packages/flutter/lib/src/material/flat_button.dart [hunk 1]
```diff
@@ -149,4 +149,25 @@ class FlatButton extends StatelessWidget {
       child: child
     );
   }
+
+  @override
+  void debugFillDescription(List<String> description) {
+    super.debugFillDescription(description);
+
+    if (onPressed != null)
```
Comment (konifar) [inline packages/flutter/lib/src/material/flat_button.dart]: Actually I'm not sure this is a correct way 🙇 
I would appreciate if you tell the better way.

**Comment by Hixie** [comment]
We've renamed debugFillDescription to debugFillProperties, and changed its semantics a bit. If you search around the code for other debugFillProperties you'll see how it works now.

**Comment by konifar** [comment]
@Hixie Thanks! I'll check it and fix PR!

**Comment by konifar** [comment]
@Hixie Changed to use `debugFillProperties` instead of `debugFillDescription`.

File: packages/flutter/lib/src/material/flat_button.dart [hunk 2]
```diff
@@ -162,4 +162,17 @@ class FlatButton extends StatelessWidget {
...
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder description) {
+    super.debugFillProperties(description);
+    description.add(new ObjectFlagProperty<VoidCallback>('onPressed', onPressed, ifNull: 'disabled'));
+    description.add(new DiagnosticsProperty<Color>('textColor', textColor, defaultValue: null));
+    description.add(new DiagnosticsProperty<Color>('disabledTextColor', disabledTextColor, defaultValue: null));
+    description.add(new DiagnosticsProperty<Color>('color', color, defaultValue: null));
+    description.add(new DiagnosticsProperty<Color>('highlightColor', highlightColor, defaultValue: null));
+    description.add(new DiagnosticsProperty<Color>('splashColor', splashColor, defaultValue: null));
+    description.add(new DiagnosticsProperty<Widget>('child', child, defaultValue: null));
```
Comment (Hixie) [inline packages/flutter/lib/src/material/flat_button.dart]: no need to list the child, we automatically add that already

**Comment by Hixie** [comment]
LGTM

**Comment by Hixie** [comment]
Thanks! Looks great.

If you remove the child line, leave a comment here, and I can land it for you.

File: packages/flutter/lib/src/material/flat_button.dart (hunk 2 above)
Comment (konifar) [reply packages/flutter/lib/src/material/flat_button.dart]: Thanks. I removed it!

**Comment by Hixie** [comment]
Thanks! I shall land this for you. Thanks again for the contribution!

**Pull request merged**